src/UI/Internal/FiltersHeaders.elm

```diff
@@ -287,6 +308,180 @@ singleTextFilterRender renderConfig applyMsg { fromFiltersMsg, index, label } ed
     editable
         |> Filters.editableWithDefault ""
         |> TextField.singlelineText editMsg label
+        |> TextField.withSize Size.extraSmall
         |> TextField.withWidth TextField.widthFull
         |> TextField.withOnEnterPressed applyMsg
         |> TextField.renderElement renderConfig
+
+
+multiTextFilterRender :
+    RenderConfig
+    -> msg
+    -> Config msg
+    -> Filters.Editable (Array String)
+    -> Element msg
+multiTextFilterRender renderConfig applyMsg { fromFiltersMsg, index, label } editableArr =
+    let
+        editMsg subIndex str =
+            fromFiltersMsg <| Filters.EditMultiText { column = index, field = subIndex, value = str }
+
+        single subIndex line =
+            line
+                |> TextField.singlelineText (editMsg subIndex) label
+                |> TextField.withSize Size.extraSmall
+                |> TextField.withWidth TextField.widthFull
+                |> TextField.withOnEnterPressed applyMsg
+                |> TextField.renderElement renderConfig
+
+        instanceArr =
+            Filters.editableWithDefault Array.empty editableArr
+    in
+    instanceArr
```

Comment:
I would remove instanceArr function and just:

```elm
editableArry
        |> Filters.editableWithDefault Array.empty
        |> Array.push ""
        |> Array.indexedMap single
        |> Array.toList
        |> Element.column [ Element.width fill, Element.spacing 8 ]
```
just to reduce the let..in functions